plutus-core/plutus-core/src/PlutusCore/Builtin/Runtime.hs

```diff
@@ -55,32 +51,15 @@ instance NoThunks (BuiltinRuntime val) where
 
     showTypeOf = const "PlutusCore.Builtin.Runtime.BuiltinRuntime"
 
--- | Determines how to unlift arguments. The difference is that with 'UnliftingImmediate' unlifting
--- is performed immediately after a builtin gets the argument and so can fail immediately too, while
--- with deferred unlifting all arguments are unlifted upon full saturation, hence no failure can
--- occur until that. The former makes it much harder to specify the behaviour of builtins and
--- so 'UnliftingDeferred' is the preferred mode.
-data UnliftingMode
-    = UnliftingImmediate
-    | UnliftingDeferred
-
 -- | A 'BuiltinRuntimeOptions' is a precursor to 'BuiltinRuntime'. One gets the latter from the
--- former by applying a function returning the runtime denotation of the builtin (either
--- '_broImmediateF' for immediate unlifting or '_broDeferredF' for deferred unlifting, see
--- 'UnliftingMode' for details) to a cost model.
-data BuiltinRuntimeOptions val cost = BuiltinRuntimeOptions
-    { _broImmediateF :: cost -> BuiltinRuntime val
-    , _broDeferredF  :: cost -> BuiltinRuntime val
-    }
+-- former by applying a function returning the runtime denotation of the builtin.
+data BuiltinRuntimeOptions val cost = BuiltinRuntimeOptions (cost -> BuiltinRuntime val)
```

Comment:
Yeah, I wasn't 100% sure what to do with this type now. I should go look at whether it existed before. Or do you think we might have more use for it in future?

Reply:
It wasn't really "options" before TBH.

Reply:
> I should go look at whether it existed before.

It didn't, but the machinery was very different too. If you can drop it just fine, do it, I don't see how it can be helpful in future.

> It wasn't really "options" before TBH.

As in "you can choose from either of these two options". I see what you mean, though.